ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,109 @@ actor_system_config:
+
+By default, the hard memory limit is set to the {{ ydb-short-name }} node process's [cgroups](https://en.wikipedia.org/wiki/Cgroups) memory limit.
+
+In environments without a cgroups memory limit, the default hard memory limit is equal to the host's total available memory. This allows the database to utilize all available resources in unrestricted environments, though it may lead to resource contention with other processes running on the same host.
+
+Additionally, the hard memory limit can be specified in the configuration. Note that the database process may still exceed this limit, so it is highly recommended to use cgroups memory limits in production environments to enforce strict memory control.
+
+Example of the `memory_controller_config` section with a specified hard memory limit:
+
+```yaml
+memory_controller_config:
+  hard_limit_bytes: 16106127360
+```
+
+### Soft memory limit {#soft-memory-limit}
+
+The soft memory limit specifies a dangerous threshold that should not be exceeded under normal conditions.
+
+If the soft limit is exceeded, {{ ydb-short-name }} starts to reduce the shared cache size to zero. Therefore, more database nodes should be added to the cluster as soon as possible or per-component memory limits should be reduced.
+
+### Target memory utilization {#target-memory-utilization}
+
+The target memory utilization specifies a threshold for memory usage that is considered optimal.
+
+Flexible cache sizes are calculated according to their limit thresholds to keep process consumption around this value.
+
+For example, in a database that consumes a little memory on query execution, caches consume memory around this threshold and other memory stays free. And if query execution consumes more memory, caches start to reduce their limits to their minimum threshold.
+
+### Per-component memory limits
+
+#### Flexible memory limits
+
+Some {{ ydb-short-name }} components have both minimum and maximum memory limit thresholds, allowing for dynamic adjustment based on current process consumption.
+
+These components are:
+
+- Shared cache
+- MemTable
+
+Each of these components' limits is dynamically recalculated every second so that each component consumes memory proportionally to its limit threshold and the total consumed memory stays around the target memory utilization.
+
+The minimum memory limit threshold for these components isn't reserved and remains free until consumed.
+
+Memory limits can be configured either in absolute bytes or as a percentage relative to the [hard memory limit](#hard-memory-limit). Using percentages is advantageous for managing clusters with nodes of varying capacities. If both absolute byte and percentage limits are specified, the memory controller uses a combination of both (maximum for lower limits and minimum for upper limits).
+
+Both the minimum and maximum thresholds should be overridden if needed; otherwise, a missing threshold will have a default value.
```

Comment:
```suggestion
#### Flexible memory limits

Some {{ ydb-short-name }} components have both minimum and maximum memory limit thresholds, allowing for dynamic adjustment based on current process consumption.

These components are:

- Shared cache
- MemTable

Each of these components' limits is dynamically recalculated every second so that each component consumes memory proportionally to its limit threshold, and the total consumed memory stays around the target memory utilization.

These components' minimum memory limit threshold isn't reserved and remains free until consumed.

Memory limits can be configured either in absolute bytes or as a percentage relative to the [hard memory limit](#hard-memory-limit). Using percentages is advantageous for managing clusters with nodes of varying capacities. If both absolute byte and percentage limits are specified, the memory controller uses a combination of both (maximum for lower limits and minimum for upper limits).

If needed, both the minimum and maximum thresholds should be overridden; otherwise, a missing threshold will have a default value.
```